### PR TITLE
Add `log_file`

### PR DESCRIPTION
**Pull request by Froloket64**
Add a simple way to redirect error output (stderr) into a file.

#7

### CODE REVIEW

File: nurses_2/app.py [hunk 1]
```diff
@@ -96,6 +96,7 @@ def __init__(
         render_interval: float=0.0,
         color_theme: ColorTheme=DEFAULT_COLOR_THEME,
         asciicast_path: Path | None=None,
+        log_file: str | None=None,
```
Comment (salt-die) [inline nurses_2/app.py:99]: Annotate this as a `Path` to be consistent with everything.

File: nurses_2/app.py [hunk 2]
```diff
@@ -136,7 +138,11 @@ def run(self):
         except asyncio.CancelledError:
             pass
         finally:
-            print(defer_stderr.getvalue(), file=sys.stderr, end="")
+            if self.log_file:
+                with open(self.log_file, "w") as log:
+                    print(defer_stderr.getvalue(), file=log, end="")
```
Comment (salt-die) [inline nurses_2/app.py:143]:
```py
if self.log_file:
    self.log_file.write_text(defer_stderr.getvalue())
```
Once a path, this method will be available.

Comment (Froloket64) [reply nurses_2/app.py:143]: Ok, but this will force the user to cast to `Path` first

**Comment by salt-die** [comment]
Should `log_file` be `log_path`? For consistency? I don't feel strongly about it.

**Comment by Froloket64** [comment]
Well, I think "file" works better semantically. It's a log FILE, not just some PATH. That's how I feel

**Comment by Froloket64** [comment]
Although... I don't even know. It **would** be more consistent with "path" after all

**Comment by salt-die** [comment]
I feel like one of `asciicast_path` or `log_file` should change to match the other. I don't think I have much of a preference. I do use `path` as an argument to some widgets, so there's that.

**Comment by Froloket64** [comment]
It just kinda sounds more like a phrase: "a log file" as opposed to "a log path". At least that is to me...

**Comment by salt-die** [comment]
Ok, I'll pull it and update docs.

**Pull request merged**